server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -640,7 +640,7 @@ public void updateShardState(
                     if (currentRouting.initializing() && currentRouting.isRelocationTarget() == false && newRouting.active()) {
                         // the cluster-manager started a recovering primary, activate primary mode.
                         replicationTracker.activatePrimaryMode(getLocalCheckpoint());
-                        ensurePeerRecoveryRetentionLeasesExist();
```

Comment:
Curious my understanding is this is a no-op?

Reply:
No, it won't be. We moved this to `postActivatePrimaryMode`